client/jetpack-cloud/sections/agency-dashboard/hooks/use-toggle-activate-monitor.tsx

```diff
@@ -24,32 +25,41 @@ export default function useToggleActivateMonitor(
 
 	const queryClient = useQueryClient();
 	const { filter, search, currentPage, sort } = useContext( SitesOverviewContext );
-
-	const { dataViewsState, showOnlyFavorites } = useContext( SitesDashboardContext );
+	const { dataViewsState, showOnlyFavorites, showOnlyDevelopmentSites } =
+		useContext( SitesDashboardContext );
 
 	const agencyId = useSelector( getActiveAgencyId );
 
-	const queryKey = isA8CForAgencies()
-		? [
-				'jetpack-agency-dashboard-sites',
-				dataViewsState.search,
-				dataViewsState.page,
-				{
-					issueTypes: getSelectedFilters( dataViewsState.filters ),
-					showOnlyFavorites: showOnlyFavorites || false,
-				},
-				dataViewsState.sort,
-				dataViewsState.perPage,
-				...( agencyId ? [ agencyId ] : [] ),
-		  ]
-		: [
-				'jetpack-agency-dashboard-sites',
-				search,
-				currentPage,
-				filter,
-				sort,
-				...( agencyId ? [ agencyId ] : [] ),
-		  ];
+	const queryKey = useMemo(
+		() =>
+			isA8CForAgencies()
+				? [
+						'jetpack-agency-dashboard-sites',
+						hashParameters( [
+							dataViewsState.search,
+							dataViewsState.page,
+							{
+								issueTypes: getSelectedFilters( dataViewsState.filters ),
+								showOnlyFavorites: showOnlyFavorites || false,
+								showOnlyDevelopmentSites: showOnlyDevelopmentSites || false,
+							},
+							dataViewsState.sort,
+							dataViewsState.perPage,
+							...( agencyId ? [ agencyId ] : [] ),
+						] ),
+				  ]
+				: [
+						'jetpack-agency-dashboard-sites',
+						hashParameters( [
+							search,
+							currentPage,
+							filter,
+							sort,
+							...( agencyId ? [ agencyId ] : [] ),
+						] ),
+				  ],
+		[ agencyId, currentPage, dataViewsState, showOnlyFavorites, showOnlyDevelopmentSites ]
```

Comment:
<img width="879" alt="Screenshot 2024-08-30 at 10 32 44 AM" src="https://github.com/user-attachments/assets/3a8dada4-c747-4cde-a6b8-75f483d83825">

Can we add these missing dependencies?

Reply:
https://github.com/Automattic/wp-calypso/pull/94023/commits/09668a27edaae31e0007cea9a7d9a8751a97f741